frontend/model/logger.js

```diff
@@ -0,0 +1,118 @@
+import sbp from '@sbp/sbp'
+import { CAPTURED_LOGS } from '~/frontend/utils/events.js'
+import CircularList from '~/shared/CircularList.js'
+
+const loggingLevels = ['debug', 'error', 'info', 'log', 'warn']
+const originalConsole = console
+const noop = (...args: any) => undefined
+
+export async function createLogger (config: Object, { getItem, removeItem, setItem }: { getItem: Function, removeItem: Function, setItem: Function }): Object {
+  const entries = new CircularList(config.maxEntries)
+  const methods = Object.fromEntries(loggingLevels.map((name) =>
+    [name, (...args) => {
+      originalConsole[name](...args)
+      captureLogEntry(logger, name, config.source, ...args)
+    }]
+  ))
+  const appLogsFilter: string[] = []
+  const consoleProxy = new Proxy({ ...methods }, {
```

Comment:
Should this be simply `new Proxy(methods, ..`? Why are we copying to a new object?

Reply:
For immutability.

```js
consoleProxy[level] = appLogsFilter.includes(level) ? methods[level] : noop
```

If it wasn't copied over, the line shown above will reassign it.